### PR TITLE
deprecate metrics pkg 1/n

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

This is the initial PR with deprecation process, here we drop docs and forward the user to the new package

In follow-up PR we will ... (suggesting one metric ~ one PR)
1. connect particular metric to `torchmetrics` - import the relevant metric from TM and replace all function body  PL metric by a call to the TM version of the same metric
2. add a deprecation warning - just after the func signature or class init add a warning
3. test deprecation warning
4. drop particular metric test

see for example:
https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/metrics/functional/classification.py#L375-L415

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/metrics/__init__.py [hunk 1]
```diff
@@ -37,3 +39,8 @@
     R2Score,
     SSIM,
 )
+
+warn(
+    "`pytorch_lightning.metrics.*` package has been renamed to `torchmetrics.*` since v1.3 and will be removed in v1.5",
```
Comment (SkafteNicki) [inline pytorch_lightning/metrics/__init__.py:44]:
```suggestion
    "`pytorch_lightning.metrics.*` module has been renamed to `torchmetrics.*` and split off to its own package (https://github.com/PyTorchLightning/metrics) since v1.3 and will be removed in v1.5",
```

**Review by SkafteNicki — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

File: docs/source/extensions/metrics.rst [hunk 2]
```diff
@@ -1,887 +1,9 @@
...
-``pytorch_lightning.metrics`` is a Metrics API created for easy metric development and usage in
-PyTorch and PyTorch Lightning. It is rigorously tested for all edge cases and includes a growing list of
-common metric implementations.
-
-The metrics API provides ``update()``, ``compute()``, ``reset()`` functions to the user. The metric base class inherits
-``nn.Module`` which allows us to call ``metric(...)`` directly. The ``forward()`` method of the base ``Metric`` class
-serves the dual purpose of calling ``update()`` on its input and simultaneously returning the value of the metric over the
-provided input.
+``pytorch_lightning.metrics`` has been moved to separate package `TorchMetrics <https://torchmetrics.readthedocs.io/>`_.
+We will preserve user compatibility for next few following release never the less we encourage users to start transition to this stand-alone package.
```
Comment (carmocca) [inline docs/source/extensions/metrics.rst:6]:
```suggestion
We will preserve compatibility for the next few releases, nevertheless, we encourage users to update to use this stand-alone package.
```

File: docs/source/extensions/metrics.rst [hunk 3]
```diff
@@ -1,887 +1,9 @@
...
 
-``pytorch_lightning.metrics`` is a Metrics API created for easy metric development and usage in
-PyTorch and PyTorch Lightning. It is rigorously tested for all edge cases and includes a growing list of
-common metric implementations.
-
-The metrics API provides ``update()``, ``compute()``, ``reset()`` functions to the user. The metric base class inherits
-``nn.Module`` which allows us to call ``metric(...)`` directly. The ``forward()`` method of the base ``Metric`` class
-serves the dual purpose of calling ``update()`` on its input and simultaneously returning the value of the metric over the
-provided input.
+``pytorch_lightning.metrics`` has been moved to separate package `TorchMetrics <https://torchmetrics.readthedocs.io/>`_.
```
Comment (carmocca) [inline docs/source/extensions/metrics.rst:5]:
```suggestion
``pytorch_lightning.metrics`` has been moved to a separate package `TorchMetrics <https://torchmetrics.readthedocs.io/>`_.
```

File: pytorch_lightning/metrics/__init__.py [hunk 4]
```diff
@@ -37,3 +39,8 @@
     R2Score,
     SSIM,
 )
+
+warn(
+    "`pytorch_lightning.metrics.*` module has been renamed to `torchmetrics.*` and split off to its own package (https://github.com/PyTorchLightning/metrics) since v1.3 and will be removed in v1.5",
```
Comment (SkafteNicki) [inline pytorch_lightning/metrics/__init__.py:44]: code checker is failing due to the line being too long
```suggestion
    "`pytorch_lightning.metrics.*` module has been renamed to `torchmetrics.*`"
    " and split off to its own package (https://github.com/PyTorchLightning/metrics)"
    " since v1.3 and will be removed in v1.5",
```

**Review by carmocca — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

**Review by ananyahjha93 — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

File: pytorch_lightning/metrics/__init__.py [hunk 5]
```diff
@@ -38,3 +40,8 @@
     SSIM,
 )
 from pytorch_lightning.metrics.retrieval import RetrievalMAP  # noqa: F401
+
+warn(
+    "`pytorch_lightning.metrics.*` module has been renamed to `torchmetrics.*` and split off to its own package"
+    " (https://github.com/PyTorchLightning/metrics) since v1.3 and will be removed in v1.5", DeprecationWarning
+)
```
Comment (carmocca) [inline pytorch_lightning/metrics/__init__.py:47]: @Borda with the fixes in #8002 this warning appears every time pl is imported. Should avoid that

Comment (Borda) [reply pytorch_lightning/metrics/__init__.py:47]: Do you mean imported in the same running session?
we can have an internal counter to make sure it is raised only once...

Comment (carmocca) [reply pytorch_lightning/metrics/__init__.py:47]: I mean the following:

```python
$ python -c 'import pytorch_lightning'
/Users/carlosmocholi/git/pytorch-lightning/pytorch_lightning/metrics/__init__.py:43: LightningDeprecationWarning: `pytorch_lightning.metrics.*` module has been renamed to `torchmetrics.*` and split off to its own package (https://github.com/PyTorchLightning/metrics) since v1.3 and will be removed in v1.5
  rank_zero_deprecation(
```

Comment (dlangerm) [reply pytorch_lightning/metrics/__init__.py:47]: I'm getting major log spam with this warning on windows and it appears there is no way to turn it off.

```
import warnings
# from pytorch_lightning.utilities.warnings import LightningDeprecationWarning
# Even specifying LightningDeprecationWarning below does not work
warnings.filterwarnings("ignore", category=DeprecationWarning) 

import pytorch_lightning
```

The above code causes the deprecation warning to print, so when I have a data loader with multiple workers, this message spams the terminal.

Comment (carmocca) [reply pytorch_lightning/metrics/__init__.py:47]: Yeah, we should remove it as it's too intrusive. Would you like to open a PR?
